File: typescript/ratio.ts
import { readFileSync, readdirSync } from "node:fs";
import { encode, decode } from "./src/index.js";
let tb = 0, tc = 0;
for (const f of readdirSync("../bench/corpus").sort()) {
  let d: Uint8Array;
  try { d = new Uint8Array(readFileSync(`../bench/corpus/${f}`)); } catch { continue; }
  if (d.length === 0) continue;
  const e = encode(d);
  const back = decode(e);
  if (back.length !== d.length || !back.every((b, i) => b === d[i])) { console.log("FAIL", f); process.exit(1); }
  tb += d.length; tc += e.length;
  console.log(f.padEnd(28), (e.length / d.length).toFixed(5));
}
console.log("TOTAL".padEnd(28), (tc / tb).toFixed(5));

File: rust/examples/ratio.rs
// Scratch harness: corpus expansion ratio of the current build, with a
// round-trip check on every file.
fn main() {
    let mut tb = 0usize;
    let mut tc = 0usize;
    let mut paths: Vec<_> = std::fs::read_dir("../bench/corpus")
        .unwrap()
        .filter_map(|e| e.ok())
        .map(|e| e.path())
        .filter(|p| p.is_file())
        .collect();
    paths.sort();
    for p in paths {
        let d = std::fs::read(&p).unwrap();
        let e = base85n::encode(&d);
        let back = base85n::decode(&e).unwrap();
        assert_eq!(back, d, "roundtrip {:?}", p);
        tb += d.len();
        tc += e.len();
        println!(
            "{:<28} {:.5}",
            p.file_name().unwrap().to_string_lossy(),
            e.len() as f64 / d.len() as f64
        );
    }
    println!("{:<28} {:.5}", "TOTAL", tc as f64 / tb as f64);
}
